### PR TITLE
Runtime: add i3 config from https://github.com/mboughaba/i3config.vim

### PR DESCRIPTION
**Pull request by hiqua**
I'm not the author of this code.

Taken from: https://github.com/mboughaba/i3config.vim

Suggested in this issue where the author approved of the idea: https://github.com/mboughaba/i3config.vim/issues/34

License: MIT, so relicensing under the Vim license should be ok.

### CODE REVIEW

**Comment by hiqua** [comment]
Can I do something to help get this merged?

**Comment by brammool** [comment]
Please ask the author of the text to OK this PR.

**Comment by hiqua** [comment]
Thanks for the feedback, I guess you mean the author of the code. @mboughaba could you give your approval here?

Unfortunately it's not clear whether they are still active, so I don't know if we'll get an OK. On the other hand, the code is under the MIT license: https://github.com/mboughaba/i3config.vim/blob/master/License so legally speaking that shouldn't be an issue.

**Comment by chrisbra** [comment]
no, because we haven't heard back from the author

**Comment by hiqua** [comment]
@chrisbra what's the reasoning behind asking the original author to OK this PR when the license itself allows it?

**Comment by chrisbra** [comment]
Even if the license allows it, it's good practice that the author is okay with the distribution of the files. And having an explicit approval is always nice.

**Comment by hiqua** [comment]
Ok, the maintenance is indeed something we won't have with these files as the author is not responsive anymore, although they were fine with the suggestion that led to this PR. I hadn't expected maintenance to be required for syntax files.

I can still try to contact the maintainer of a fork, e.g. @JosefLitos would you be interested in merging your fork https://github.com/JosefLitos/vim-i3config to this repository and maintaining it like you're currently doing on your repository, so that everyone can benefit?

If there's no fork author that's willing to do it, I'll see if I can maintain the code myself. I'm not proficient at vimscript by any means, but 300 rarely used lines of code is also not something I can't handle. I'll post again once it's clear there's no other way.

**Comment by hiqua** [comment]
@JosefLitos yes I'm in a similar situation. Just open a new PR if you're interested, otherwise I'll do it myself 2 months if nobody else has stepped up by then.

**Comment by mg979** [comment]
Personally I use a much simpler syntax file for i3:
```vim
" Vim syntax file
" Language: i3wm config file

" Quit when a (custom) syntax file was already loaded
if exists("b:current_syntax")
  finish
endif

runtime syntax/conf.vim

syn keyword i3Command   bindsym bindcode exec gaps exec_always floating_modifier assign for_window
syn match   i3Command   '^\s*\%(set\>\|font\>\|mode\>\|bar\>\)'
syn match   i3Mod       '\s\+\$\w\+\>'

hi def link i3Command   Keyword
hi def link i3Mod       PreProc

let b:current_syntax = "i3"
```

The file @hiqua proposed is quite complex and if the original author doesn't maintain it, who will? If there is no maintainer, imho it's better to stick with something very basic (as the one here).

**Comment by mg979** [comment]
@hiqua Another problem is that setting too many autocommands for a specific filetype (that interests only a minority of users) slows down start up, you should at least use patterns like `*/i3/config` instead of many autocommands for all possible paths.

**Comment by hiqua** [comment]
@mg979 sure, but you didn't answer me regarding a possible interest in opening your own PR with the snippet you quoted. I'm not the original author.

**Comment by mg979** [comment]
I understood @chrisbra meant above that this PR wouldn't be a problem? Personally I prefer minimal defaults, to be extended with more complex stuff if preferred. I have no problems in submitting a PR if others agree with including a simpler syntax file.

**Comment by hiqua** [comment]
Given that:
* others do not seem to want to open a new PR
* maintenance should be lightweight according to https://github.com/vim/vim/pull/7969#issuecomment-944078402

I'll add myself as the maintainer of this code instead of the original author, and I can at least commit to being reactive to issues related to this code. I'll comment again once it's ready for review.

**Comment by brammool** [comment]
Mentioning the original author and adding someone else as (active) maintainer is fine.
We have several files where the author or previous maintainer is not responsive, and having an active maintainer is more important.

**Comment by hiqua** [comment]
Done!

**Comment by brammool** [comment]
In the ftplugin file please set b:undo_ftplugin
Please add a test for the filetype detection in src/testdir/test_filetype.vim

**Comment by hiqua** [comment]
Done.

**Comment by brammool** [comment]
The patterns are too specific, we can just match */i3/config and */sway.config.  I don't think that will give false matches.
I'll make it like that, no need to update the PR.

**Comment by brammool** [comment]
That also fixes the test failure.  I'll make a patch for the filetype detection, the other files will be in the next runtime update.

**Comment by hiqua** [comment]
Thanks a lot!

File: runtime/ftplugin/i3config.vim [hunk 1]
```diff
@@ -0,0 +1,12 @@
+" Vim syntax file
+" Language: i3 config file
+" Original Author: Mohamed Boughaba <mohamed dot bgb at gmail dot com>
+" Maintainer: Quentin Hibon
+" Version: 0.4
+" Last Change: 2019-03-23 21:54
+
+if exists("b:did_ftplugin") | finish | endif
+let b:did_ftplugin = 1
+let b:undo_ftplugin = ""
```
Comment (dkearns) [inline runtime/ftplugin/i3config.vim:10]: @brammool might have handled this but `undo_ftplugin` should be set to 'undo' the change to `commentstring`.

```vim
let b:undo_ftplugin = "setlocal commentstring<"
```

Comment (brammool) [reply runtime/ftplugin/i3config.vim:10]: Yes, I changed it in my copy.  Please also update the original.  I also made some tweaks to the syntax file.

Comment (hiqua) [reply runtime/ftplugin/i3config.vim:10]: Sorry, do you mean that this PR still needs to be updated @brammool?

Comment (brammool) [reply runtime/ftplugin/i3config.vim:10]: You appear to quote a header where there is an "Original Author" entry in the header, but what I have only has the "Maintainer" entry with Mohamed Boughab.  Please check what the correct header is.

Comment (hiqua) [reply runtime/ftplugin/i3config.vim:10]: I see that the headers are not the same everywhere, I'll open a new PR to fix this. The correct one is in `runtime/ftplugin/i3config.vim`

Comment (hiqua) [reply runtime/ftplugin/i3config.vim:10]: https://github.com/vim/vim/pull/9467

**Pull request closed without merging**